internal/cache/controller_obj_cache.go

```diff
@@ -2554,6 +2541,51 @@ func validateAndConfigureSeGroup(client *clients.AviClient) bool {
 		return false
 	}
 
+	// Indulge in Marker based SEGroup searching only when user input is empty.
+	// SE Group selection priority
+	// 1. User input
+	seGroupToUse := lib.GetSEGNameEnv()
+
+	// 2. Marker based
+	if seGroupToUse == "" {
```

Comment:
It looks like this feature is only relevant for `advancedL4`, in such a case having a top level filter on advancedL4 will optimise the code.